mock/task_executor.go

```diff
@@ -35,107 +35,128 @@ func NewExecutorRequest(ch string, fn string, args []string, isSignedInvoke bool
-	cert, err := hex.DecodeString(batchRobotCert)
-	if err != nil {
-		return nil, fmt.Errorf("failed to decode hex string batchRobotCert: %w", err)
+	if len(resp) != 1 {
+		return nil, fmt.Errorf("execute no signed invoke failed: expected 1 response, got %d", len(resp))
 	}
-	w.ledger.stubs[r.Channel].SetCreator(cert)
 
-	var args []string
-	if r.IsSignedInvoke {
-		args, _ = w.sign(r.Method, r.Channel, r.Args...)
-	}
+	return resp[0].BatchTxEvent.GetResult(), nil
+}
+
+func (w *Wallet) TaskExecutorRequest(channel string, requests ...ExecutorRequest) ([]ExecutorResponse, error) {
+	tasks := make([]*proto.Task, len(requests))
+	for i, r := range requests {
+		if r.Channel != channel {
+			return nil, fmt.Errorf("common tasks channel '%s' does not match to request channel '%s'", channel, r.Channel)
+		}
+		var args []string
+		if r.IsSignedInvoke {
+			args = w.SignArgs(r.Channel, r.Method, r.Args...)
+		} else {
+			args = r.Args
+		}
 
-	task := &proto.Task{
-		Id:     strconv.FormatInt(rand.Int63(), 10),
-		Method: r.Method,
-		Args:   args,
+		task := &proto.Task{
+			Id:     strconv.FormatInt(rand.Int63(), 10),
+			Method: r.Method,
+			Args:   args,
+		}
+
+		tasks[i] = task
 	}
 
-	bytes, err := pb.Marshal(&proto.ExecuteTasksRequest{Tasks: []*proto.Task{task}})
+	batchResponse, err := w.TasksExecutor(channel, tasks)
 	if err != nil {
-		return nil, fmt.Errorf("failed to marshal tasks ExecuteTasksRequest: %w", err)
+		return nil, err
 	}
 
-	// do invoke chaincode
-	peerResponse, err := w.ledger.doInvokeWithPeerResponse(r.Channel, txIDGen(), core.ExecuteTasks, string(bytes))
+	batchEvent, err := w.fetchBatchEvent(channel)
 	if err != nil {
-		return nil, fmt.Errorf("failed to invoke method %s: %w", core.ExecuteTasks, err)
+		return nil, err
+	}
+	responseMap := make(map[string]*proto.TxResponse)
+	for _, response := range batchResponse.GetTxResponses() {
+		responseMap[string(response.GetId())] = response
 	}
+	executorResponses := make([]ExecutorResponse, 0)
+	for _, batchTxEvent := range batchEvent.GetEvents() {
+		txResponse, ok := responseMap[string(batchTxEvent.GetId())]
+		if !ok {
+			return nil, fmt.Errorf("could not find response for event %v", batchTxEvent.GetId())
+		}
 
-	if peerResponse.GetStatus() != http.StatusOK {
-		return nil, fmt.Errorf("failed to invoke method %s, status: '%v', message: '%s'", core.ExecuteTasks, peerResponse.GetStatus(), peerResponse.GetMessage())
+		if responseErr := txResponse.GetError(); responseErr != nil {
+			return nil, errors.New(responseErr.GetError())
+		}
+		executorResponses = append(executorResponses, ExecutorResponse{
+			TxResponse:   txResponse,
+			BatchTxEvent: batchTxEvent,
+		})
 	}
 
-	var batchResponse proto.BatchResponse
-	err = pb.Unmarshal(peerResponse.GetPayload(), &batchResponse)
+	return executorResponses, nil
+}
+
+func (w *Wallet) TasksExecutor(channel string, tasks []*proto.Task) (*proto.BatchResponse, error) {
+	// setup creator
+	cert, err := hex.DecodeString(batchRobotCert)
```

Comment:
why batchRobotCert?

Reply:
thanks, this part of code is deprecated, i removed this code